src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -361,9 +361,11 @@ object desugar {
         }
         companionDefs(parent, applyMeths ::: unapplyMeth :: defaultGetters)
       }
-      else if (defaultGetters.nonEmpty)
-        companionDefs(anyRef, defaultGetters)
-      else Nil
+      else {
```

Comment:
This code was rewritten, it used to have more complicated logic between the commits. I have no problem changing it back.